pkg/virt-operator/resource/generate/components/crds.go

```diff
@@ -51,27 +51,47 @@ var (
 	PreserveUnknownFieldsFalse       = false
 )
 
-func patchValidation(crd *extv1beta1.CustomResourceDefinition) error {
+func addFieldsToVersions(versions []extv1.CustomResourceDefinitionVersion, fields ...interface{}) error {
+	for i := range versions {
+		for _, field := range fields {
+			switch v := field.(type) {
+			case []extv1.CustomResourceColumnDefinition:
+				versions[i].AdditionalPrinterColumns = v
+			case *extv1.CustomResourceSubresources:
+				versions[i].Subresources = v
+			case *extv1.CustomResourceValidation:
+				versions[i].Schema = v
+			default:
+				return fmt.Errorf("cannot add field of type %T to a CustomResourceDefinitionVersion", v)
+			}
+		}
+	}
+	return nil
+}
+
+func patchValidation(crd *extv1.CustomResourceDefinition) error {
 	name := crd.Spec.Names.Singular
 
-	crd.Spec.PreserveUnknownFields = &PreserveUnknownFieldsFalse
+	crd.Spec.PreserveUnknownFields = PreserveUnknownFieldsFalse
 	validation, ok := CRDsValidation[name]
 	if !ok {
 		return nil
 	}
-	crvalidation := extv1beta1.CustomResourceValidation{}
+	crvalidation := extv1.CustomResourceValidation{}
 	err := k8syaml.NewYAMLToJSONDecoder(strings.NewReader(validation)).Decode(&crvalidation)
 	if err != nil {
 		return fmt.Errorf("Couldn't decode validation for %s, %v", name, err)
 	}
-	crd.Spec.Validation = &crvalidation
+	if err = addFieldsToVersions(crd.Spec.Versions, &crvalidation); err != nil {
```

Comment:
I don't think we area always going to want to add all openapi spec values to every version. Since this is the version of the custom resource that we are serving. for example `virtualmachineinstance` if we decide to add or deprecate a field between versions that should be shown in the crd.